### DISCUSSION TITLE
world map component

### DISCUSSION THREAD

**Discussion by tobehn** [Budibase/budibase#2739] [Ideas]
As a app developer i would love to see a world map component in the designer. 
Basically a simple leaflet map (https://leafletjs.com/)

Geo data (coordinates) could be bound from a Data Provider including a Repeater.
The coodinates would render a simple marker on the map and a Tooltip could provide further information and a variable internal link to whereever i want to route to (internally).

![grafik](https://user-images.githubusercontent.com/40893107/134710182-ed3781cc-87c2-4c51-b14c-e40c2cccfd89.png)

A svelete skleteon for leaflet could be found here: https://github.com/dimfeld/svelte-leaflet-demo/tree/master/skeleton
A detailed video about the integration: https://www.youtube.com/watch?v=-klB-EocorE

Of course they describe a very advanced way of a datavisualisation layer on top of the map. I really only would suggest to be able to set markers via stored data an attach a few information to it similar to card component.

I would be happy to assist in any way possible.

In my opinion there wouldn't be an extra datatype for geodata. Perhaps the bound data could just be of datatype Number validated against a certain format/definitions.

**Comment by gvlx** [comment]
Hi,

There SHOULD be a new type for geodata as this can be a point or a poligon. And if you want  to use it in a GIS-capable database, you need to pack data in a way those DB understand it.

I see several ways to create a map, assuming they are svelt-based:
- https://github.com/beyonk-adventures/svelte-mapbox?ref=madewithsvelte.com
- https://imfeld.dev/writing/leaflet_with_svelte
- https://github.com/beyonk-adventures/svelte-mapbox

There are some ways to embed a map in BB as well, but they require the import of JS libraries and CSS, which is not feasible:
- http://www.visionfortech.com/2016/12/solution-for-how-to-embed-or-include-openstreetmap-into-HTML.html
- https://zyzo.github.io/gsoc/fossasia/2015/05/03/gsoc-map-picker-task.html
- https://www.thinkingondata.com/embedding-a-leaflet-map-on-wordpress/

Right now, the only option to add a map onto BB is by embedding in a component, but you cannot extract any data from it, so their use is very limited (you can use it to _show_ a location):
- https://www.embed-leaflet.com/
- https://www.embedgooglemap.net/
- https://www.embed-map.com/

I can even create a single HTML file which picks points from a map, but it cannot be embedded as it is:
[index.zip](https://github.com/Budibase/budibase/files/7661782/index.zip)

**Reply by gvlx** [reply]
What do you think @mjashanks ?
Do you see it in your roadmap any time soon?

**Reply by mjashanks** [reply]
It's definitely important and must get built. However, there are a lot of things that fit this description! I don't know when we will have it. The noisier this discussion gets, the higher priority it will become!

**Comment by kwjones** [comment]
I also see a benefit in having a map component.  I've been evaluating low-code platforms for IoT projects and that is a must have to be considered in any application where location visualization is required.  Nice product by the way!

**Comment by gzqn** [comment]
I would like to be able to nest locations at a (optionable) series of scales. I would also like to be able to define the shape of those locations  rather than be forced to use gps locations (too labour intensive) or some type of grid patterning (insufficient detail).

**Comment by mituBarua** [comment]
#help  Is there anyone can help me implementing map in budibase?? please...
